sdk/gramine/0002-helloworld-with-Nemesis-in-Gramine.patch

```diff
@@ -0,0 +1,269 @@
+diff --git a/CI-Examples/helloworld/helloworld.manifest.template b/CI-Examples/helloworld/helloworld.manifest.template
+index 64c908e1..205860da 100644
+--- a/CI-Examples/helloworld/helloworld.manifest.template
++++ b/CI-Examples/helloworld/helloworld.manifest.template
+@@ -6,6 +6,9 @@ loader.log_level = "{{ log_level }}"
+ 
+ loader.env.LD_LIBRARY_PATH = "/lib"
+ 
++loader.insecure__use_cmdline_argv = true
++loader.insecure__disable_aslr = true
++
+ fs.mounts = [
+   { path = "/lib", uri = "file:{{ gramine.runtimedir() }}" },
+   { path = "/helloworld", uri = "file:helloworld" },
+@@ -13,6 +16,8 @@ fs.mounts = [
+ 
+ sgx.debug = true
+ sgx.nonpie_binary = true
++sgx.preheat_enclave = true
++sgx.thread_num = 4
+ 
+ sgx.trusted_files = [
+   "file:{{ gramine.libos }}",
+diff --git a/pal/src/host/linux-sgx/host_ecalls.c b/pal/src/host/linux-sgx/host_ecalls.c
+index 201a0a27..f97ea21b 100644
+--- a/pal/src/host/linux-sgx/host_ecalls.c
++++ b/pal/src/host/linux-sgx/host_ecalls.c
+@@ -6,6 +6,185 @@
+ #include "pal_ecall_types.h"
+ #include "pal_rpc_queue.h"
+ 
```

Comment:
Thanks for the code! Ideally, for future maintenance, this chunk of code that does the actual single-stepping would be in a separate C file in the `sdk/gramine` directory, so that it would have proper version control etc. 

That is: the patches should ideally be only the smallest changes needed to link into SGX-Step and then the actual attack logic can be maintained and version controlled as normal C files. This has the advantage that the attack code is more easy to read and understand and extend, and, most importantly, the changes to the target runtime are minimal and can hopefully be ported easily across Gramine versions :)

Concretely I suggest:
1. moving the single-stepping logic from `host_ecalls.c` to `sdk/gramine/aep.c` or so and only keep in the patch file the minimal setup before the first ecall
2. changing the gramine build system in the patch to also include this `../aep.c` file (or alernatively, it should be possible to link it into libsgxstep instead)

(Also related, in the future, I'd like to properly refactor the single-stepping abstraction issue #2  such that all the code duplication with `app/bench/main.c` is avoided )